### DISCUSSION TITLE
AWS self deployment

### DISCUSSION THREAD

**Discussion by mike12345567** [Budibase/budibase#986] [Ideas]
<p align="center">
<img src="https://user-images.githubusercontent.com/4407001/104210849-41406c00-542b-11eb-9acf-7d645a055edc.png"  width="128">
</p>

Budibase now supports self hosting with the help of [Docker](https://www.docker.com/)/[Docker compose](https://docs.docker.com/compose/) but we want to keep making it easier to build your ideas with our tools. As part of this we plan to introduce more self hosting deployment options, one option for this is support for AWS (Amazon Web Services).

If you would be interested in a simple AWS deployment of Budibase this is the place to talk about it! We would love to know your use case for this and why you would prefer this over any other deployment option.

**Comment by sladerose** [comment]
We'd love to get more information on how to deploy to AWS.

**Reply by mjashanks** [reply]
Hi @sladerose.

If you want to run on AWS, you can do it today. You need to create a VM, access the shell, and follow the [installation instuctions](https://docs.budibase.com/getting-started). Internally, we run an instance on AWS, using an EC2 t2-medium instance.

We have plans to support a one-click deploy from AWS marketplace over the next few months.

**Reply by murilobd** [reply]
Out of curiosity, do you run couchdb in the same instance as you're running budbase-server? Or do you have a different instance only for couchdb?

**Reply by mike12345567** [reply]
Hi @murilobd - for all of our installation options right now CouchDB is ran on the same instance, but it would be possible/easy enough to run in a separate instance if desired.

Internally we run a very large CouchDB cluster across many EC2 instances, but this isn't really needed unless you have hundreds of thousands of apps!

**Reply by murilobd** [reply]
Thanks @mike12345567 !
And, talking about MinIO and S3, would be possible to entirely replace MinIO for S3 under a docker (or budi-cli) installation? I've seen some examples on how to do it using k8s, but didn't find a solution for docker-compose

**Reply by mike12345567** [reply]
Hi @murilobd - we don't expose the environment variables in the `.env` file for Docker-compose right now, we plan on documenting how to update all the various environment variables across all the different environments and their purposes, but for now you can look at the server (app image) and worker (worker image) environment variable files [here](https://github.com/Budibase/budibase/blob/develop/packages/server/src/environment.js) and [here](https://github.com/Budibase/budibase/blob/develop/packages/worker/src/environment.js). If you update the MinIO URL, access and secret key environment variables for the two services in your docker-compose file you should be able to connect to S3.

**Comment by nmabhinandan** [comment]
EC2 instance + CouchDB cluster + S3.

**Reply by MrChadMWood** [reply]
Should one use the `MINIO_*` environment variables when specifying credentials for S3?

**Comment by farzadpanahi** [comment]
Any new updates on this idea? How about deploying to AWS Fargate?

**Reply by mjashanks** [reply]
No updates! Looking back, I said "Next few months" :/

It's going to be important for us but can't give you a date.

**Reply by mjashanks** [reply]
Happy to accept a contribution if you have skills in this area.

**Reply by CristovaoMendes** [reply]
Do you have news about this project?

**Comment by lake1984** [comment]
Budibase sounds incredible. Longtime user of: Access, SFDC, LongJump, Sheets, TeamDesk, etc. I will try your hosted version while I wait for a 1click AWS option - thanks!

**Comment by aledt** [comment]
Our use case is fairly simple - we have a number of external services that we're working with that provide data to our team over a range of APIs. We want to expose that data and functionality to our operations team and analysts without developing an entire application around them so Budibase sounds ideal. 

However, this is the finanvial services sector so ALL of the APIs we work with are completely locked down. We have a setup in AWS that allows us to access them but there's no way we can allow multi-tenant cloud services to access them. As a result, we need it self-hosted, but we also want it to fit into our existing deployment architecture which is primarily ECS Fargate - which means proper containerisation. If the app architecture supports that and we like Budibase I'm happy to put some time on our side towards a clean deployment mechanism for somethign like that and put that back in the community.

**Comment by kpturner** [comment]
Bit disappointed to see no updates here for nearly a year.  Has anyone good a good way of deploying budibase in AWS Lambda and an indication of how the data is persisted across multiple instances of the lambda?

OR just a simple ECS setup?

**Comment by Marcello1099** [comment]
I have deployed Budibase on AWS Elastic Beanstalk using Amazon Elastic File System.

Check out the setup and configurations here: [Budibase on AWS Elastic Beanstalk](https://github.com/Marcello1099/budibase-aws-eb).

Your feedback and contributions are welcome to refine and improve this method!

**Reply by gnani-sekar** [reply]
Hi, This worked like a charm. Thank you very much.

Just that I am now struggling to enable HTTPS. Can you please guide through on how to get the HTTPS installed with custom subdomain.
